### PR TITLE
Improvements for automatically generated release notes

### PR DESCRIPTION
**Pull request by PokhodenkoSA**
Added `.github/release.yml` file according to this documentation
https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes

Also, assigned labels to PRs for generating better release notes.
Updated CHANGELOG.md with example of generated release notes.

Target to `release0.17` branch.

### CODE REVIEW

**Comment by PokhodenkoSA** [comment]
```diff
diff --git a/CHANGELOG.md b/CHANGELOG.md
--- CHANGELOG.md
+++ CHANGELOG.md
@@ -6,8 +6,9 @@
 
 ## [0.17.0] - 2021-11-xx
 
 ### Added
+
 * Use Python 3.9 [public CI] by @PokhodenkoSA in https://github.com/IntelPython/numba-dppy/pull/574
 * Use `NUMBA_DPPY_DEBUG` for debugging GDB tests by @PokhodenkoSA in https://github.com/IntelPython/numba-dppy/pull/578
 * Preliminary support Numba 0.55 (master branch) by @PokhodenkoSA in https://github.com/IntelPython/numba-dppy/pull/583
 * Workflow for manually running tests using Numba PRs by @PokhodenkoSA in https://github.com/IntelPython/numba-dppy/pull/586
@@ -16,8 +17,9 @@
 * Update to dpctl 0.11 by @PokhodenkoSA in https://github.com/IntelPython/numba-dppy/pull/595
 * Update to dpnp 0.9 by @PokhodenkoSA in https://github.com/IntelPython/numba-dppy/pull/599
 * Improve the documenatation landing page by @diptorupd in https://github.com/IntelPython/numba-dppy/pull/601
 * Clean up README by @diptorupd in https://github.com/IntelPython/numba-dppy/pull/604
+
 ### Fixed
 * Restrict dpctl to 0.10.* for release 0.16 by @1e-to in https://github.com/IntelPython/numba-dppy/pull/590
 * Fix upload from release branch by @1e-to in https://github.com/IntelPython/numba-dppy/pull/596
 * Unskip tests passing with dpnp 0.9.0rc1 by @PokhodenkoSA in https://github.com/IntelPython/numba-dppy/pull/606

```

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,19 +4,25 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [0.17.0] - 2021-10-xx
+## [0.17.0] - 2021-11-xx
```
Comment (PokhodenkoSA) [inline CHANGELOG.md:8]:
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,19 +4,25 @@ All notable changes to this project will be documented in this file.
...
 
 ### Added
```
Comment (PokhodenkoSA) [inline CHANGELOG.md:9]:
```suggestion
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,19 +4,24 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [0.17.0] - 2021-10-xx
-
+## [0.17.0] - 2021-11-xx
```
Comment (PokhodenkoSA) [inline CHANGELOG.md:7]:
```suggestion
## [0.17.0] - 2021-11-xx

```

**Review by 1e-to — APPROVED** [review]

**Comment by PokhodenkoSA** [comment]
Merged. No need to wait internal CI.

**Pull request merged**